### DISCUSSION TITLE
Use myst-parser in a sphinx extension to parse md to docutils nodes

### DISCUSSION THREAD

**Discussion by ewjoachim** [executablebooks/MyST-Parser#880] [Q&A] [answered]
I'm trying to refactor a part of [Sphinx-GitHub-Changelog](https://github.com/ewjoachim/sphinx-github-changelog), a sphinx extension that generates a changelog based on GitHub releases.

I can have either the GFM or the HTML of a GitHub release body (whichever is best). I'd like to use Myst-Parser to turn those into docutils nodes that my sphinx directive should return.

Does Myst-Parser expose an API to do that? I've spent some time on the doc and couldn't find it. There's the `MystParser.parse()`, but I don't really have a `document` to provide (my directive is expected to return nodes, not write them on the document directly) and even if I did, it would need to have myst config, which I don't know how to pass. I'm (as usual) confused by the Docutils API.

Anyone has an idea ?

**Comment by holmboe** [comment]
Is the Python code in https://myst-parser.readthedocs.io/en/latest/docutils.html#single-page-builds of any help to you?

**Reply by ewjoachim** [reply]
Thanks for the suggestion! I've read it, but didn't play with it. From the first glance, I don't see how to output docutils nodes.

**Reply by agoose77** [reply, marked as answer]
`myst-parser` renders MarkdownIt tokens and populates an existing docutils node (document). See https://github.com/executablebooks/MyST-Parser/blob/d3d7fbb6a4fd2ecd12c7a29d7de902a869897eb9/myst_parser/mdit_to_docutils/base.py#L100 for the base-class of the renderer that is used.

Does this provide a helpful place to start? :)

**Reply by ewjoachim** [reply]
Thank you for your answer, I'll do my best to get going from there.
